File: TripKitAndroid/src/main/java/com/skedgo/tripkit/LineSegment.java
package com.skedgo.tripkit;

import com.skedgo.tripkit.common.util.TripKitLatLng;
import com.skedgo.tripkit.routing.Trip;

/**
 * Represents a segment of a polyline denoting a {@link Trip}.
 */
public final class LineSegment {
    public final TripKitLatLng start;
    public final TripKitLatLng end;
    public final int color;
    public final String tag;

    public LineSegment(TripKitLatLng start, TripKitLatLng end, int color, String tag) {
        this.start = start;
        this.end = end;
        this.color = color;
        this.tag = tag;
    }

    public enum Tag {
        SHAPE,
        STREET
    }
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/LineSegment.kt
package com.skedgo.tripkit

import com.skedgo.tripkit.common.util.TripKitLatLng

/**
 * Represents a segment of a polyline denoting a [Trip].
 */
class LineSegment @JvmOverloads constructor(
    val start: TripKitLatLng,
    val end: TripKitLatLng,
    val color: Int,
    val tag: String,
    val tripUuid: String? = null,
    val segmentId: Long? = null
) {
    enum class Tag {
        SHAPE,
        STREET
    }
}
